codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessage.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.util.AbstractReferenceCounted;
+import io.netty.util.CharsetUtil;
+import io.netty.util.internal.StringUtil;
+
+import java.util.List;
+
+/**
+ * Redis Message for <a href="http://redis.io/topics/protocol">RESP (REdis Serialization Protocol)</a>
+ */
+public class RedisMessage extends AbstractReferenceCounted {
+
+    private static final int MAX_PEEK_BULK_STRING = 20;
+
+    private RedisMessageType type;
+
+    // content(type!=arrays) or children(type==arrays).
+    private ByteBuf content;
```

Comment:
How about splitting `RedisMessage` into particular subtypes? Like Array, integer (long), simple string and a type containing just the `ByteBuf`? Reflecting the message type as real type makes the usage simpler and omits the need to check whether the content fits to the message type.